speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/SQLDatabaseStorage.cpp

```diff
@@ -311,64 +305,59 @@ void SQLDatabaseStorage::CreateTables()
         "    \n"
         "    CONSTRAINT gear_id_primary_key PRIMARY KEY (intervention_id),\n"
         "    CONSTRAINT gear_key_is_intervention FOREIGN KEY (intervention_id) REFERENCES Intervention(intervention_id)\n"
-        ")")
+        ")");
 }
 
-/// @brief Inserts the data that stays the same during a trial. Includes participant, blackbox, environment, settings, and trial
-/// @return trialId
-int SQLDatabaseStorage::InsertInitialData(std::ifstream& p_inputFile)
+/// @brief Inserts meta data about the trial. Includes participant, blackbox, environment, settings, and trial.
+/// @return trialId The auto-incremented id corresponding to this trial.
+int SQLDatabaseStorage::InsertMetaData(std::ifstream& p_inputFileStream)
 {
     std::string values;
 
-    // participant
+    // Store participant meta-data
     std::string participantId;
 
-    READ_INPUT(p_inputFile, participantId);
-
-    EXECUTE(INSERT_IGNORE_INTO("Participant", "participant_id", participantId))
-
-    std::string trialDateTime;
-
-    READ_INPUT(p_inputFile, trialDateTime);
+    READ_LINE(p_inputFileStream, participantId);
+    EXECUTE(INSERT_IGNORE_INTO("Participant", "participant_id", participantId));
 
-    // blackbox
+    // Store blackbox meta-data
     std::string blackboxFileName;
     std::string blackboxVersionDateTime;
     std::string blackboxName;
 
-    READ_INPUT(p_inputFile, blackboxFileName);
-    READ_INPUT(p_inputFile, blackboxVersionDateTime);
-    READ_INPUT(p_inputFile, blackboxName);
+    READ_LINE(p_inputFileStream, blackboxFileName);
+    READ_LINE(p_inputFileStream, blackboxVersionDateTime);
+    READ_LINE(p_inputFileStream, blackboxName);
 
     values = "'" + blackboxFileName + "','" + blackboxVersionDateTime + "','" + blackboxName + "'";
 
-    EXECUTE(INSERT_IGNORE_INTO("Blackbox", "filename, version, name", values))
-    EXECUTE_QUERY("SELECT blackbox_id FROM Blackbox WHERE filename = '" + blackboxFileName + "' AND version = '" + blackboxVersionDateTime + "'")
-
-    int blackboxId;
-    GET_INT_FROM_RESULTS(blackboxId)
+    int blackboxId = -1;
+    EXECUTE(INSERT_IGNORE_INTO("Blackbox", "filename, version, name", values));
+    GET_INT_FROM_QUERY(blackboxId,
+        "SELECT blackbox_id FROM Blackbox "
+        "WHERE filename = '" + blackboxFileName + "' AND version = '" + blackboxVersionDateTime + "'")
```

Comment:
Doesn't this allow for sql injections based on the blackboxfilename?

Reply:
Could be fixed by removing spaces as DROP TABLE would become DROPTABLE